handlers/handlers.go

```diff
@@ -0,0 +1,178 @@
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			http.Error(w, "post not found", http.StatusNotFound)
+			return
+		}
+
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	resp := map[string]db.Post{"post": post}
+
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		http.Error(w, "failed to write response", http.StatusInternalServerError)
+	}
+}
+
+// UpdatePost updates a particular post
+func (h *Handlers) UpdatePost(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var payload struct {
+		Title string `json:"title"`
+		Body  string `json:"body"`
+	}
+
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		http.Error(w, "invalid id", http.StatusBadRequest)
+		return
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		http.Error(w, "invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	p := db.Post{
+		ID:    id,
+		Title: payload.Title,
+		Body:  payload.Body,
+	}
+
+	if err := h.db.UpdatePost(r.Context(), id, p); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			http.Error(w, "post not found", http.StatusNotFound)
+			return
+		}
+
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	if err != nil {
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	resp := map[string]db.Post{
+		"post": p,
+	}
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		http.Error(w, "failed to write response", http.StatusInternalServerError)
+	}
+}
+
+// CreatePost creates a new post
+func (h *Handlers) CreatePost(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var payload struct {
+		Title string `json:"title"`
+		Body  string `json:"body"`
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		http.Error(w, "invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	p := db.Post{
+		Title: payload.Title,
+		Body:  payload.Body,
+	}
+
+	if err := h.db.CreatePost(r.Context(), &p); err != nil {
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	resp := map[string]db.Post{"post": p}
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		http.Error(w, "failed to write response", http.StatusInternalServerError)
+	}
+}
+
+// DeletePost deletes a post
+func (h *Handlers) DeletePost(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		http.Error(w, "invalid id", http.StatusBadRequest)
+		return
+	}
+
+	if err := h.db.DeletePost(r.Context(), id); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			http.Error(w, "post not found", http.StatusNotFound)
+			return
+		}
+
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
```

Comment:
I've never seen this code used before 😅. What does it mean?

Reply:
Here is an example of where it is used: https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event. I use it when a request is successful, and when I don't have anything in the response body, especially when deleting in REST